pkg/kubectl/cmd/cmd_test.go

```diff
@@ -45,6 +49,166 @@ import (
 	"k8s.io/kubernetes/pkg/util/strings"
 )
 
+type mockFactory struct {
+	factory cmdutil.Factory
```

Comment:
It would be easier to just embed this, then you would not need to re-implement the methods.  This isn't doing any mocking yet.